backend/src/main/java/com/bakdata/conquery/models/query/statistics/Histogram.java

```diff
@@ -0,0 +1,132 @@
+package com.bakdata.conquery.models.query.statistics;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import it.unimi.dsi.fastutil.doubles.Double2ObjectFunction;
+import it.unimi.dsi.fastutil.doubles.DoubleArrayList;
+import it.unimi.dsi.fastutil.doubles.DoubleList;
+import lombok.Data;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Simple implementation of a histogram.
+ * First and last bin serve as potential overflow bins.
+ * <p>
+ * Bin labels are of real values and not partitions, this can make entries potentially non-contiguous, but ensures readable values.
+ */
+@Data
+@Slf4j
+public class Histogram {
+
+
+	private final Node[] nodes;
+	private final Node zeroNode, underflowNode, overflowNode;
+	private final double lower, upper;
+	private final double width;
+
+	private int total;
+
+	public static Histogram zeroCentered(double lower, double upper, int expectedBins, boolean contains0, boolean round) {
+
+		final double width = round ? Math.ceil((upper - lower) / expectedBins) : (upper - lower) / expectedBins;
+
+		final double newLower;
+
+		if (lower == 0) {
+			newLower = 0;
+		}
+		else if (contains0) {
+			// We adjust slightly downward so that we have even sized bins, that meet exactly at zero (which is tracked separately)
+			newLower = Math.signum(lower) * width * Math.ceil(Math.abs(lower) / width);
+		}
+		else {
+			newLower = lower;
+		}
+
+		final double newUpper = newLower + width * expectedBins;
+
+		final Node[] nodes = IntStream.range(0, expectedBins)
+									  .mapToObj(index -> new Node(newLower + width * index, newLower + width * (index + 1)))
+									  .toArray(Node[]::new);
+
+
+		return new Histogram(nodes, new Node(0, 0), new Node(Double.NEGATIVE_INFINITY, newLower), new Node(newUpper, Double.POSITIVE_INFINITY), newLower, newUpper, width);
+
+	}
+
+	public void add(double value) {
+		total++;
+
+		if (value == 0d) {
+			zeroNode.add(value);
```

Comment:
Okay, es ersten zeros gezählt, aber warum?

Reply:
Anweisung vom Chef.

(Aber auch einfach, weil null in den meisten Situationen ein spezieller Fall ist, e.g. 0€ Kosten, oder 0 Arzt-Besuche etc.)